db/migrate/20231117182638_fill_approve_unless_allowed_groups_based_on_deprecated_settings.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+class FillApproveUnlessAllowedGroupsBasedOnDeprecatedSettings < ActiveRecord::Migration[7.0]
+  def up
+    approve_unless_trust_level_raw =
+      DB.query_single(
+        "SELECT value FROM site_settings WHERE name = 'approve_unless_trust_level'",
+      ).first
+
+    # Default for old setting is TL0, we only need to do anything if it's been changed in the DB.
+    if approve_unless_trust_level_raw.present?
+      # Matches Group::AUTO_GROUPS to the trust levels.
+      approve_unless_allowed_groups = "1#{approve_unless_trust_level_raw}"
+
+      # Data_type 20 is group_list
+      DB.exec(
+        "INSERT INTO site_settings(name, value, data_type, created_at, updated_at)
+        VALUES('approve_unless_allowed_groups', :setting, '20', NOW(), NOW())",
```

Comment:
I am wondering if we should preserve `created_at`?

Reply:
Normally for an update yes, but we are inserting here so there isn't an existing created_at. We should only be inserting if there isn't already a record.